File: servpy/misc/GraphDb.py
import os
# from flask import g
from neo4j import GraphDatabase

# from app import app

graph_auth_un = os.environ.get('NEO4J_DB_USER', "neo4j")
graph_auth_pw = os.environ.get('NEO4J_DB_PW', "neo4j")
graph_auth = (graph_auth_un, graph_auth_pw)

host = os.environ.get('NEO4J_DB_HOST', "localhost")
port = os.environ.get('NEO4J_DB_PORT', 7474)
bolt_port = os.environ.get('NEO4J_DB_PORT_BOLT', 7687)
graph_endpoint = 'http://{}:{}/db/data/transaction/commit'.format(host, port)
bolt_endpoint = 'bolt://{}:{}'.format(host, bolt_port)

class GraphDb:
	def init(self):
		self.driver = GraphDatabase.driver(bolt_endpoint, auth=graph_auth, encrypted=False)

	@classmethod
	def instance(cls):
		"""Singleton like accessor to instantiate backend object"""

		if not hasattr(cls, "_instance"):
			cls._instance = cls()
			cls._instance.init()

		return cls._instance

	def getDb(self):
		return self.driver

'''
def get_db():
	with app.app_context():
		if not hasattr(g, 'neo4j_db'):
			g.neo4j_db = driver
		return g.neo4j_db
'''

File: servgo/graphdb.go
package main

import (
	"os"
	"strconv"

	"github.com/neo4j/neo4j-go-driver/neo4j"
)

var graphAuthUser = getEnv("NEO4J_DB_USER", "neo4j")
var graphAuthPw = getEnv("NEO4J_DB_PW", "neo4j")

var host = getEnv("NEO4J_DB_HOST", "localhost")
var port = getEnvInt("NEO4J_DB_PORT", 7474)
var boltPort = getEnvInt("NEO4J_DB_PORT_BOLT", 7687)
var graphEndpoint = "http://" + host + ":" + strconv.Itoa(port) + "/db/data/transaction/commit"
var boltEndpoint = "bolt://" + host + ":" + strconv.Itoa(boltPort)

var db = initGraphDb()

func initGraphDb() neo4j.Driver {
	config := func(c *neo4j.Config) {
		c.Encrypted = false
	}
	driver, err := neo4j.NewDriver(boltEndpoint, neo4j.BasicAuth(graphAuthUser, graphAuthPw, ""), config)
	if err != nil {
		return nil
	}
	return driver
}

func getEnv(key string, defaultVal string) string {
	if value, exists := os.LookupEnv(key); exists {
		return value
	}
	return defaultVal
}

func getEnvInt(key string, defaultVal int) int {
	if value, exists := os.LookupEnv(key); exists {
		if iValue, err := strconv.Atoi(value); err != nil {
			return iValue
		}
	}
	return defaultVal
}
